core/src/main/java/org/apache/iceberg/CatalogUtil.java

```diff
@@ -129,6 +129,35 @@ private static void deleteFiles(FileIO io, Set<ManifestFile> allManifests) {
         });
   }
 
+  /**
+   * Get constructor for Catalog from impl. Impl must be valid class name and have zero-arg constructor.
+   *
+   * <p>We try to get the constructor for impl and if we fail we check for common error patterns. Specifically:
+   * a ClassNotFoundException for an invalid impl String and a NoClassDefFoundError if impl fails to get initialized.
+   * Otherwise the original exception is passed up and the error is assumed to be a missing no-args constructor.
+   *
+   * @param impl catalog implementation full class name
+   * @return constructor for catalog implementation impl
+   * @throws IllegalArgumentException if no-arg constructor not found or error during initialization
+   */
+  private static DynConstructors.Ctor<Catalog> getCatalogConstructor(String impl) {
```

Comment:
Nit: use of `get` when a more descriptive verb would be better. For example, `find` implies that it may not be found.